File: ArtDrop/src/main/java/hr/tvz/artdrop/artdropapp/model/Challenge.java
package hr.tvz.artdrop.artdropapp.model;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "challenge")
public class Challenge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "created_by")
    private Long createdBy;

    @Column(nullable = false)
    private String title;

    @Column(length = 2000)
    private String description;

    @Column(length = 1000)
    private String quote;

    @Enumerated(EnumType.STRING)
    @Column(length = 30)
    private ChallengeKind kind;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private ChallengeStatus status;

    private String theme;

    @Column(name = "cover_image_url", length = 1000)
    private String coverImageUrl;

    @Column(name = "starts_at")
    private LocalDateTime startsAt;

    @Column(name = "ends_at")
    private LocalDateTime endsAt;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "challenge", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ChallengeSubmission> submissions = new ArrayList<>();
}

File: artdropapp-frontend/src/types/challenge.ts
export type ChallengeStatus = 'UPCOMING' | 'ACTIVE' | 'ENDED'

export interface SubmissionThumbnail {
  submissionId: number
  artworkId: number
  title: string
  imageUrl: string
  imageAlt: string
  artistDisplayName: string | null
  artistSlug: string | null
}

export interface Challenge {
  id: number
  title: string
  description: string | null
  quote: string | null
  isFeatured: boolean
  status: ChallengeStatus | null
  theme: string | null
  coverImageUrl: string | null
  startsAt: string | null
  endsAt: string | null
  submissionCount: number
  submissions: SubmissionThumbnail[]
  viewerHasEntry: boolean
  viewerEntryArtworkId: number | null
}
